src/object.d

```diff
@@ -1964,15 +1964,15 @@ V[K] dup(T : V[K], K, V)(T* aa)
     return (*aa).dup;
 }
 
-auto byKey(T : V[K], K, V)(T aa) pure nothrow @nogc
+auto byKey(T : V[K], K, V)(T aa) pure nothrow @nogc @trusted
 {
     import core.internal.traits : substInout;
 
     static struct Result
     {
         AARange r;
 
-    pure nothrow @nogc:
+    pure nothrow @nogc @trusted:
```

Comment:
> To be fair, this is also applies to std.range.front for normal arrays when compiling in release mode without bounds checking, and that's marked as safe.

But that's `-boundscheck=off`'s fault. It flat-out breaks `@safe`. An `@safe`/`@trusted` function may be unsafe with the switch, but it must be safe without it.